File: app/src/main/java/com/ne1c/gitteroid/utils/RxSchedulersFactory.java
package com.ne1c.gitteroid.utils;

import rx.Scheduler;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public class RxSchedulersFactory {
    public Scheduler io() {
        return Schedulers.io();
    }

    public Scheduler androidMainThread() {
        return AndroidSchedulers.mainThread();
    }
}

File: app/src/main/kotlin/com/ne1c/gitteroid/utils/RxSchedulersFactory.kt
package com.ne1c.gitteroid.utils

import rx.Scheduler
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

open class RxSchedulersFactory {
    open fun io(): Scheduler {
        return Schedulers.io()
    }

    open fun androidMainThread(): Scheduler {
        return AndroidSchedulers.mainThread()
    }
}
